src/common/models/model.h

```diff
@@ -67,11 +69,12 @@ class FModel
 	virtual ~FModel();
 
 	virtual bool Load(const char * fn, int lumpnum, const char * buffer, int length) = 0;
-	virtual int FindFrame(const char * name, bool nodefault = false) = 0;
-	virtual void RenderFrame(FModelRenderer *renderer, FGameTexture * skin, int frame, int frame2, double inter, int translation, const FTextureID* surfaceskinids) = 0;
+	virtual int FindFrame(const char* name, bool nodefault = false) = 0;
+	virtual void RenderFrame(FModelRenderer *renderer, FGameTexture * skin, int frame, int frame2, double inter, int translation, const FTextureID* surfaceskinids, const TArray<VSMatrix>& animationData) = 0;
 	virtual void BuildVertexBuffer(FModelRenderer *renderer) = 0;
 	virtual void AddSkins(uint8_t *hitlist, const FTextureID* surfaceskinids) = 0;
 	virtual float getAspectFactor(float vscale) { return 1.f; }
+	virtual const TArray<VSMatrix>* AttachAnimationData() = 0;
```

Comment:
Is there any reason to make this function pure virtual? A virtual functions that returns `nullptr` makes more sense here.